lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -4,20 +4,79 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
-
 #include "hermes/Platform/Intl/PlatformIntl.h"
 
 #import <Foundation/Foundation.h>
 
+NSString *u16StringToNSString(std::u16string src) {
+  auto size = src.size();
+  auto cString = (const unichar *)src.c_str();
+  return [NSString stringWithCharacters:cString length:size];
+}
+
+std::u16string nsStringToU16String(NSString *src) {
+  auto size = src.length;
+  std::u16string result;
+  result.resize(size);
+  [src getCharacters:(unichar *)&result[0] range:NSMakeRange(0, size)];
+  return result;
+}
+
 namespace hermes {
 namespace platform_intl {
 
+// Implementation of https://tc39.es/ecma402/#sec-canonicalizelocalelist
 vm::CallResult<std::vector<std::u16string>> getCanonicalLocales(
     vm::Runtime *runtime,
     const std::vector<std::u16string> &locales) {
-  return std::vector<std::u16string>{u"fr-FR", u"es-ES"};
-}
-
+  // 1. If locales is undefined, then
+  // Return a new empty List.
+  if (locales.size() == 0) {
+    return std::vector<std::u16string>();
+  }
```

Comment:
We don't actually need this check, note that the below code will generate an empty vector if `locales` is empty.
```suggestion
```